### <b>Table of Content</b>

0. Import functions

1. Load data

2. Create Bollinger Bands

### <b>0. Import functions</b>

In [1]:
import sys
import os
from pathlib import Path

### <b>1. Load data</b>

In [44]:
# Details of the source file in G Drive
base_url = "https://drive.google.com/file/d/"
file_id = "1MiunF_O8eNWIcfaOA4PVm668RN7FgLNA0a6U4LWf5Bk"
file_url = f"{base_url}{file_id}"
file_name = "2020Q1Q2Q3Q4-2021Q1.xlsx"

# Details of local directories
root_dir = sys.path[0]
download_dir = Path(root_dir) / "data"
file_path = download_dir / file_name

if os.path.exists(file_path):
    print(f"File {file_name} already exists in {download_dir}.")
else:
    if os.getcwd() != download_dir:
        os.chdir(download_dir)
    print(f"File {file_name} is not found in {download_dir}.")
    !gdown 1MiunF_O8eNWIcfaOA4PVm668RN7FgLNA0a6U4LWf5Bk

if os.getcwd() != root_dir:
    os.chdir(root_dir)

File 2020Q1Q2Q3Q4-2021Q1.xlsx is not found in c:\Users\Admin\Documents\GitHub\Apziva\FAxAtx2ZjFTGstEu\data.


Downloading...
From (uriginal): https://drive.google.com/uc?id=1MiunF_O8eNWIcfaOA4PVm668RN7FgLNA0a6U4LWf5Bk
From (redirected): https://docs.google.com/spreadsheets/d/1MiunF_O8eNWIcfaOA4PVm668RN7FgLNA0a6U4LWf5Bk/export?format=xlsx
To: c:\Users\Admin\Documents\GitHub\Apziva\FAxAtx2ZjFTGstEu\data\2020Q1Q2Q3Q4-2021Q1.xlsx

0.00B [00:00, ?B/s]
77.8kB [00:00, 292kB/s]
111kB [00:00, 207kB/s] 
128kB [00:00, 253kB/s]


In [46]:
import pandas as pd
from openpyxl import load_workbook

data_df = {}
wb = load_workbook(file_path)
sheet_names = wb.sheetnames
for sheet_name in sheet_names:
    sheet= wb[sheet_name]
    data = [[cell.value for cell in row] for row in sheet.iter_rows()]
    data_df[sheet_name] = pd.DataFrame(data[1:], columns=data[0])

data_df

{'Russia - Sberbank Rossii PAO (S':                     Date           Price                Open             High  \
 0    2021-03-31 00:00:00          291.02               294.0           294.42   
 1    2021-03-30 00:00:00          293.25              294.86           295.72   
 2    2021-03-29 00:00:00           293.3              289.72           294.09   
 3    2021-03-26 00:00:00          291.22              288.63           292.75   
 4    2021-03-25 00:00:00          286.66               286.5           287.03   
 ..                   ...             ...                 ...              ...   
 307  2020-01-09 00:00:00          257.99               259.4           261.76   
 308  2020-01-08 00:00:00          259.15              253.57           259.15   
 309  2020-01-06 00:00:00           253.9              254.75           254.84   
 310  2020-01-03 00:00:00           255.0              255.99           258.19   
 311      Highest: 296.07  Lowest: 172.15  Difference: 123.92  

In [47]:
data_df[sheet_names[2]]

Date         Price              Open           High  \
0    2021-03-31 00:00:00          3.75              3.74           3.82   
1    2021-03-30 00:00:00          3.73              3.71           3.76   
2    2021-03-29 00:00:00           3.7              3.69           3.73   
3    2021-03-28 00:00:00          3.71              3.76           3.76   
4    2021-03-25 00:00:00          3.71              3.62           3.78   
..                   ...           ...               ...            ...   
301  2020-01-08 00:00:00          4.41              4.19           4.59   
302  2020-01-06 00:00:00          4.28              4.16           4.44   
303  2020-01-05 00:00:00          4.41               4.7            4.7   
304  2020-01-02 00:00:00          4.81              4.81           4.87   
305        Highest: 4.92  Lowest: 2.40  Difference: 2.52  Average: 3.55   

                  Low     Vol.  Change %  
0                3.71    1.57M    0.0054  
1                3.67    3.43M    0.0081  
2                3.68    1.25M   -0.0027  
3                3.68    2.34M    0.0000  
4                3.59    2.50M    0.0137  
..                ...      ...       ...  
301              4.15    3.52M    0.0304  
302              4.12    2.52M   -0.0295  
303              4.33    3.76M   -0.0832  
304              4.78  241.60K   -0.0021  
305  Change %: -22.20     None       NaN  

[306 rows x 7 columns]

2. Create Bollinger Bands for 2020 for each company




Typical values used:

Short term: 10 day moving average, bands at 1.5 standard deviations. (1.5 times the standard dev. +/- the SMA)

Medium term: 20 day moving average, bands at 2 standard deviations.

Long term: 50 day moving average, bands at 2.5 standard deviations.